src/ashell/jerry-code.c

```diff
@@ -74,6 +120,100 @@ static void javascript_print_value(const jerry_value_t value)
     jerry_port_console("\n");
 }
 
+static int read_file(const char *file_name, char **file_buf)
+{
+    fs_file_t *fp = fs_open_alloc(file_name, "r");
+
+    if (fp == NULL)
+        return 0;
+
+    ssize_t size = fs_size(fp);
+    if (size == 0) {
+        comms_printf("[ERR] Empty file (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    *file_buf = (char *)zjs_malloc(size);
+    if (*file_buf == NULL) {
+        comms_printf("[ERR] Not enough memory for (%s)\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    ssize_t brw = fs_read(fp, *file_buf, size);
+
+    if (brw != size) {
+        comms_printf("[ERR] Failed loading code %s\n", file_name);
+        fs_close_alloc(fp);
+        return 0;
+    }
+
+    return size;
```

Comment:
You're leaking the fp descriptor on the success path, need to fs_close_alloc. This would be a great time to use ZVAL's cleanup trick:

```c
fs_file_t __attribute__ ((__cleanup__(fs_close_alloc)) *fp = fs_open_alloc(...
```

(Or it might pass a `fs_file_t **` to the function in which case you'd have to write a wrapper for fs_close_alloc.) Anyway, that would let you remove all four calls to fs_close_alloc (including the missing one).